src/boot/boot.c

```diff
@@ -1673,11 +1680,36 @@ static size_t config_find_entry(Config *config, const char16_t *pattern) {
         return IDX_INVALID;
 }
 
+static bool sysfail_process(Config *config) {
+        SysFailType sysfail_type;
+
+        assert(config);
+
+        sysfail_type = sysfail_check();
+        if (sysfail_type == SYSFAIL_NO_FAILURE)
+                return false;
+
+        /* Store reason string in LoaderSysFailReason EFI variable */
+        const char16_t *reason_str = sysfail_get_error_str(sysfail_type);
+        if (reason_str)
+                (void) efivar_set_str16(MAKE_GUID_PTR(LOADER), u"LoaderSysFailReason", reason_str, 0);
+
+        config->sysfail_occured = true;
+
+        return true;
+}
+
 static void config_select_default_entry(Config *config) {
         size_t i;
 
         assert(config);
 
+        i = config_find_entry(config, config->entry_sysfail);
+        if (config->sysfail_occured && i != IDX_INVALID) {
```

Comment:
please check sysfail_occured first, and then go looking for a suitable entry. seems surprising to do it the other way round

Reply:
@poettering sorry, missed that, fixed. thanks!